lib/tesla/middleware/path_params.ex

```diff
@@ -34,9 +54,50 @@ defmodule Tesla.Middleware.PathParams do
 
   defp build_url(url, nil), do: url
 
+  defp build_url(url, %_{} = params), do: build_url(url, Map.from_struct(params))
+
   defp build_url(url, params) do
-    Regex.replace(@rx, url, fn match, key ->
-      to_string(params[String.to_existing_atom(key)] || match)
-    end)
+    params
+    |> Enum.map(fn {name, value} -> {to_string(name), value} end)
+    |> Enum.sort_by(fn {name, _} -> String.length(name) end, :desc)
```

Comment:
I am a bit confused by the need of sorting 🤔

Reply:
Because this maintains Phoenix-style parameters, and we need to support `/user/:id/post/:id_post` and replacing in reverse order ensures that `[id_post: 3, id: 1]` works. There *may* be an issue when you just put `[id: 1]` (I think that might do `/user/1/post/1_post`; I will need to add a test and figure out how to fix it.)

Reply:
Yeah. That's going to be a problem. I need to figure out how to address this particular issue. I think that I’m going to need to have separate steps for handling Phoenix-style parameters vs OpenAPI style parameters.

Reply:
I would branch between OpenAPI and Phoenix-style as soon as we can, and figure out the complexity independently.

I am OK copying Phoenix source code, but as long as OpenAPI is outside of the complexity.